tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -461,6 +489,17 @@ def get_all_test_cases():
             yield generate_compat_test(cipher=cipher, sig_alg=sig_alg, named_group=named_group,
                                        server=server, client=client)
 
+        # Generate Hello Retry Request  compat test cases
+        for client, server, client_named_group, server_named_group in \
+            itertools.product(CLIENT_CLASSES.keys(),
+                              SERVER_CLASSES.keys(),
+                              NAMED_GROUP_IANA_VALUE.keys(),
+                              NAMED_GROUP_IANA_VALUE.keys()):
+            if client_named_group != server_named_group:
+                yield generate_hrr_compat_test(client=client, server=server,
+                                               cert_sig_alg="ecdsa_secp256r1_sha256",
+                                               client_named_group=client_named_group,
+                                               server_named_group=server_named_group)
```

Comment:
```suggestion
        # Generate Hello Retry Request  compat test cases
        for client, server, server_named_group in \
            itertools.product(CLIENT_CLASSES.keys(),
                              SERVER_CLASSES.keys(),
                              NAMED_GROUP_IANA_VALUE.keys()):
           client_name_groups = [ name_group for name_group in NAMED_GROUP_IANA_VALUE.keys() if name_group != server_name_group ]
           yield generate_hrr_compat_test(client=client, server=server,
                                          cert_sig_alg="ecdsa_secp256r1_sha256",
                                          client_named_group=client_named_groups[0],
                                          server_named_group=server_named_group)
```

I do not think we should go through all possible client_named_group.

Reply:
Seems there is no standard, but here we want to cover enough group pairs.

Reply:
This does not cost much more to cover all pairs thus better to do it I'd say.

Reply:
Yes.